datastore/common/common.go

```diff
@@ -1,11 +1,118 @@
 package common
 
 import (
+	"fmt"
+	"log"
 	"os"
+	"strconv"
+	"strings"
+	"time"
 
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+var (
+	dynamicTime bool // whether the valid time range is considered dynamic or static
+	loTimeSecs, hiTimeSecs int64
+)
+
+// initValidTimeRange initializes dynamicTime, loTimeSecs, and hiTimeSecs from environment
+// variables DYNAMICTIME, LOTIME, and HITIME.
+//
+// The valid time range is defined like this if dynamicTime is true:
+//
+//     [now - loTimeSecs, now - hiTimeSecs]
+//
+// where {lo|hi}TimeSecs is converted directly as integer seconds from {LO|HI}TIME.
+//
+// If dynamicTime is false, the valid time range is defined like this:
+//
+//     [loTimeSecs, hiTimeSecs]
+//
+// where {lo|hi}TimeSecs is either converted directly as integer seconds from {LO|HI}TIME, or
+// converted indirectly from {LO|HI}TIME represented as an ISO-8601 datetime of the exact form
+// 2023-10-10T00:00:00Z.
+//
+// NOTE: if any errors are encountered, warnings will be printed and fallback values will be used.
+func initValidTimeRange() {
+	dynTime0 := strings.ToLower(Getenv("DYNAMICTIME", "true"))
+	dynamicTime = ((dynTime0 != "false") && (dynTime0 != "no") && (dynTime0 != "0"))
+
+	// define function that derives a valid int64 time range component from env.var
+	// 'name', with a default value of 'defaultVal', and with the option of
+	// expressing the value as an ISO-8601 datetime as a fallback.
+	getSecs := func(name string, defaultVal int64, allowIso8601 bool) int64 {
```

Comment:
- Go question, why do you use an inline function here? I don't see any closures.

Reply:
The function is defined as an anonymous function since it's only used inside the enclosing function. Exactly for the same reason as local variables are only accessible inside the scope of the function body. So this is all about code readability.